lib/expression.ts

```diff
@@ -68,12 +68,15 @@ class ExpressionTranspiler extends base.TranspilerBase {
 
       case ts.SyntaxKind.PropertyAccessExpression:
         var propAccess = <ts.PropertyAccessExpression>node;
-        this.visit(propAccess.expression);
         if (propAccess.name.text === 'stack' &&
             this.hasAncestor(propAccess, ts.SyntaxKind.CatchClause)) {
+          // Handle `e.stack` accesses in catch clauses by mangling to `e_stack`.
+          // FIXME: Use type checker/FacadeConverter to make sure this is actually Error.stack.
+          this.visit(propAccess.expression);
           this.emitNoSpace('_stack');
```

Comment:
There is a test for this behaviour (in `test/statement_test.ts`), note that I'm only rewording the comment and slightly refactoring the code.